pytest_perf/runner.py

```diff
@@ -78,7 +79,18 @@ def __init__(self, extras=(), deps=(), control=None):
         spec = f'[{",".join(extras)}]' if extras else ''
         self.stack = contextlib.ExitStack()
         self.control_env = self._setup_env(upstream_url(spec, control), *deps)
-        self.experiment_env = self._setup_env(f'.{spec}', *deps)
+        target = os.environ.get("PYTEST_PERF_WHEEL_TARGET")
+        if target:
+            if spec:
+                # Assuming wheel name is normalised we can derive the package name:
+                name, _, _ = os.path.basename(target).partition("-")
```

Comment:
I'd prefer to load the name from the wheel metadata rather than relying on the filename. Even better would be if this project didn't have to do this work and could rely on pip to "install this wheel with these extras".

Reply:
Ok, so I did some digging and I found that `pip` allows installing either local distribution files or `.` directly with extras, but all the other kind of installations need to use the `name[extras] @ url` syntax.

Based on that I changed the code to always use either `.` or a local distribution file, even for the upstream url. That makes it less error prone when trying to find the package name.